.circleci/scripts/git-diff-develop.js

```diff
@@ -0,0 +1,76 @@
+const { promisify } = require('util');
+const exec = promisify(require('child_process').exec);
+const fs = require('fs');
+const path = require('path');
+
+
+async function fetchWithDepthIncrement(depth) {
+  try {
+    await exec(`git fetch --depth ${depth} origin develop`);
+    await exec(`git fetch --depth ${depth} origin ${process.env.CIRCLE_BRANCH}`);
+    return true;
+  } catch (error) {
+    console.error(`Failed to fetch with depth ${depth}:`, error.message);
+    return false;
+  }
+}
+
+async function gitDiffWithRetry() {
+  const depths = [1, 10, 100];
+  let diffOutput = '';
+
+  for (const depth of depths) {
+    try {
+      console.log(`Attempting git diff with depth ${depth}...`);
+      await fetchWithDepthIncrement(depth);
+
+      const { stdout: diffResult } = await exec(`git diff --name-only origin/develop...${process.env.CIRCLE_BRANCH}`);
```

Comment:
Nit: Perhaps we could do this in two parts - find the merge base, then diff once we have it. This may be faster, and would avoid having to parse strings to determine if we found what we needed.

This command can tell us what the merge-base is: `git merge-base origin/HEAD HEAD` (we use it elsewhere in CI)
And if it can't be found, it returns with a non-zero exit code. 

e.g.

```typescript
function fetchUntilMergeBaseFound() {
  const depths = [1, 10, 100];
  for (const depth of depths) {
    console.log(`Attempting git diff with depth ${depth}...`);
    await fetchWithDepthIncrement(depth);

    try {
      await exec(`git merge-base origin/HEAD HEAD`);
      return;
    } catch (error) {
      // `git merge-base` returns an exit code of 1 when no match is found
      if (error.code === 1) {
        console.error(`Error 'no merge base' encountered with depth ${depth}. Incrementing depth...`);
      } else {
        throw error;
      }
    }
  }
  await exec(`git fetch --unshallow origin develop`);
}
```

Then we can call this once and get the diff, without repeating the diff command multiple times

Reply:
thank you for the suggestion, Mark. I've split it up in two functions now